lib/html_parser.dart

```diff
@@ -800,3 +804,45 @@ class HtmlParser {
     return false;
   }
 }
+
+class DetailsWidget extends StatefulWidget {
+  final ParseNodeFunction parseFunc;
+  final dom.Element node;
+  DetailsWidget(this.node, this.parseFunc);
+  _DetailsWidgetState createState() => _DetailsWidgetState();
+}
+
+class _DetailsWidgetState extends State<DetailsWidget> {
+  dom.Node _summaryNode;
+  bool _isOpen = false;
+
+  void initState() {
+    super.initState();
+    _summaryNode = widget.node.children
+        .firstWhere((el) => el.localName == "summary", orElse: () => null);
+    _isOpen = _summaryNode != null
+        ? _summaryNode.attributes.containsKey("open")
+        : false;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        GestureDetector(
+          onTap: () {
+            setState(() {
+              _isOpen = !_isOpen;
+            });
+          },
+          child: Text(_summaryNode != null ? _summaryNode.text : ""),
```

Comment:
The `<summary>` tag allows for elements other than text to be included as children. If this could be switched to a `Wrap` with its children as the children of the `<summary>` node, that would be great.

Reply:
Also, the standard implementation of the `<summary>` tag includes a dropdown arrow to the left of the content of the `<summary>` tag. If you could add that, I can mark this as a fully supported tag. If not, the tag will be listed under partially supported and I will work on adding the dropdown arrow later.

Reply:
think i can do this